03week/towersOfHanoi.js

```diff
@@ -19,30 +19,59 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+function movePiece(startStack, endStack) {
   // Your code here
-
+  // Moves the pieces from one stack to another
+  stacks[endStack].push(stacks[startStack].pop());
 }
 
-function isLegal() {
+function isLegal(startStack, endStack) {
   // Your code here
-
+  // Legal moves
+  if (
+    stacks[startStack][stacks[startStack].length - 1] === undefined &&
+    stacks[startStack][stacks[endStack].length - 1] === undefined
+  ) {
+    return false;
+  } else if (
+    stacks[startStack][stacks[startStack].length - 1] <
+      stacks[endStack][stacks[endStack].length - 1] ||
+    stacks[endStack][stacks[endStack].length - 1] === undefined
+  ) {
+    return true;
+  } else {
+    console.log("Invalid move.");
+    return false;
+  }
 }
 
 function checkForWin() {
   // Your code here
-
+  // Checks for a win
+  if (stacks.b.length === 4) {
+    return true;
+  } else {
+    return false;
+  }
 }
 
 function towersOfHanoi(startStack, endStack) {
   // Your code here
-
+  // Runs the game
+  let newStart = startStack.toLowerCase().trim();
+  let newEnd = endStack.toLowerCase().trim();
+  if (movePiece(newStart, newEnd)) {
+    if (isLegal(startStack, endStack)) {
+      movePiece(newStart, newEnd);
+      checkForWin();
```

Comment:
checkForWin(), returns true or false. 
That should be in an if condition, and if the result is true, you should let the user know.